mautic/mautic
# Contact Integration details extra data

Added extraData to Contact Integration details to enhance display possibilities.

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | yes
| New feature?                           | yes
| Deprecations?                          | yes
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 




#### Description:
A simple way to add extra data to the Contact's integration details.




#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a contact
3. Create an eventListener (or edit one), and Subscribe to `CoreEvents::VIEW_INJECT_CUSTOM_TEMPLATE`
4. Add extra-data to the integrations
```php
$integrations[] = [
                'integration'           => 'Mautic',
                'integration_entity'    => 'Integration',
                'integration_entity_id' => '1234',
                'date_added'            => (new \DateTime())->format('Y-m-d H:i:s'),
                'last_sync_date'        => (new \DateTime())->format('Y-m-d H:i:s'),
                'extra-data'            => [
                    'This is extra data: 23%'
                ],
            ];

            $vars['integrations'] = $integrations;

            $event->setVars($vars);
```
5. Check the extra data in the Contact/Integrations View
![Screenshot 2021-08-08 at 16 57 39](https://user-images.githubusercontent.com/79137416/128636432-da464083-ed0c-4646-8fb5-f7212aace1dd.png)
